packages/sst/src/constructs/SsrSite.ts

```diff
@@ -513,6 +553,47 @@ export abstract class SsrSite extends Construct implements SSTConstruct {
     };
   }
 
+  /** @internal */
+  protected get ssrOrigin() {
+    if (this._singletonSsrOrigin) return this._singletonSsrOrigin;
+
+    // TODO
+    console.log({ support: this.supportsStreaming() });
+    const { timeout } = this.props;   
+    const fnUrl = this.serverLambdaForRegional!.addFunctionUrl({
+      authType: FunctionUrlAuthType.NONE,
+      invokeMode: this.supportsStreaming()
+        ? InvokeMode.RESPONSE_STREAM
+        : undefined,
+    });
+    
+    
+    this._singletonSsrOrigin = new HttpOrigin(Fn.parseDomainName(fnUrl.url), {
+      readTimeout:
+        typeof timeout === "string"
+          ? toCdkDuration(timeout)
+          : CdkDuration.seconds(timeout),
+    });
+    return this._singletonSsrOrigin;
+  }
```

Comment:
This is a change from how similar resources have been instantiated, but resources like `S3Origin` are the same independent of Edge or Regional deployment. That's not the case for this `HttpOrigin` since it only exists for Regional deployment. This method allows for essentially JIT creation of resources once they are utilized the first time, and then references the singleton for all future requests.